koku/koku/middleware.py

```diff
@@ -260,10 +262,38 @@ def create_customer(account, org_id, request_method):
                     customer.save()
                     UNIQUE_ACCOUNT_COUNTER.inc()
                     LOG.info("Created new customer from account_id %s and org_id %s.", account, org_id)
-        except IntegrityError:
-            customer = Customer.objects.filter(org_id=org_id).get()
+                    return customer
+                else:
+                    LOG.warning(
+                        log_json(
+                            msg="Customer object not saved",
+                            request_method=request_method,
+                            account_id=account,
+                            org_id=org_id,
+                        )
+                    )
+                    return None
```

Comment:
Why is this change here? We still return the Customer object for `GET` and `HEAD` requests, but this change undoes that. We *intentionally do not save a customer* for `GET` and `HEAD` requests.